src/_posts/2020-06-01-signs-your-software-is-rotting.md

```diff
@@ -0,0 +1,125 @@
+
+## Struggling to keep pace with the demands of the business ##
+
+Closely linked to the previous point, when a team starts to struggle to keep pace with the demands of the organisation that is also an indication that the software codebase could be in an unhealthy state. For example, perhaps the domain modelling in the codebase is no longer a good fit and requires a new model to be designed in order to facilitate the needs of the business requirements. When a software team is unable to keep pace with the demands of the organisation there is a serious problem. It means that the organisation is at risk of losing any competitive edge it has in the market as it will take it longer to ship new features to its customers. This leaves the organisation open to be overtaken by competitors who are able to deliver new features to customers much faster and on a continuous basis.
+
+## Change Failure Rate ##
+
+It’s all well and good deploying your software to production frequently, but if these deployments result in breaking changes being introduced that impact your customer’s ability to use your product then that’s not good at all. We want to be deploying new changes into production regularly, but not at the cost of quality. Change Failure Rate is a metric that was introduced by the DevOps Research and Assessment team (DORA) and is a measure of how often a deployment to production results in a failure of some kind being introduced. A high or increasing change failure rate suggests that our software is not as healthy as it should be and is likely missing adequate quality gates on the path to production. I wrote about this in a previous article, which you can find here : https://codurance.com/2020/03/12/achieving-stability-and-speed-in-software-delivery.
+
+## Declining Software Metrics ##
+
+Software metrics are essentially measures of certain quality attributes of a codebase. It’s important to note that I’d advise against looking for absolute numbers across any of these attributes. Instead, it’s far more useful to focus on the trend over time. A decline tells us that our software has become unhealthy in a certain area and we should take action to resolve it, to avoid software rot setting in. 
+
+There are a number of metrics that can be collected about a software codebase, the following list is therefore not exhaustive but a collection of those I have personally found useful in the teams I have worked with
```

Comment:
I would list out the metrics you're going to discuss purely to reinforce the messaging.